### PR TITLE
Feature/only print update message when local danger is outdated

### PR DESCRIPTION
**Pull request by JuanitoFatas**
Hi 👋 , This PR implements [Update error message #184](https://github.com/danger/danger/issues/184).

When an `Danger::DSL::Error` is raised:

* When local danger is outdated, prints the update danger message.
* When local danger is not outdated / latest version file not found, not print the update message.
* Outdated is checked with https://rubygems.org

<details>
<summary>staled information</summary>

Outdated? By looking at the latest danger version file (save at `~/.danger_version `).
The latest danger version file is added / updated when the danger library is loaded (`Danger.setup!`).
</details>

--

Some questions 🙋

* [x] When user's home directory is not writable 🤔 / not a directory (😱)
  ❗ will raise an `Danger::HomeDirectoyError` (**BREAKING BEHAVIOUR** 🚸).
  ~~- [ ] Should we fail?~~

  - [x] Or not fail, skip silently? => Should not fail for this tiny feature

~~* [ ] Should we only update the latest danger version file in a period of time?~~

~~* [ ] The latest danger version file location is okay?~~
  ~~- [ ] `~/.danger_version `~~

~~* [ ] The source file location is okay?~~
  - [x] `lib/danger/clients/rubygems_client.rb`

  ~~- [ ] `lib/danger/services/home_keeper.rb`~~

~~* [ ] [cross-platform home directory](https://ci.appveyor.com/project/orta/danger-47jiu/build/1.0.363) (dig rubygems/rubygems source code, maybe?)~~

Thanks! 🌈

### CODE REVIEW

**Comment by orta** [comment]
> When user's home directory is not writable 🤔 / not a directory (😱)
> ❗ will raise an Danger::HomeDirectoryError (BREAKING BEHAVIOUR 🚸).

Seems fine, Danger's nearly always running on CI, [ish](https://github.com/danger/danger/issues/382) 

> Should we fail?
> Or not fail, skip silently?
> Should we only update the latest danger version file in a period of time?

We should just continue, this feature is definitely not important enough (in the grand scheme of why someone would use Danger ) to warrant breaking someone's build

>  The latest danger version file location is okay?

`~/.danger_version` seems fine to me - yes

> The source file location is okay?
> - lib/danger/clients/rubygems_client.rb
> - lib/danger/services/home_keeper.rb

Yep

>  cross-platform home directory (dig rubygems/rubygems source code, maybe?)

Yep

**Comment by hanneskaeufler** [comment]
Looking epic, I wondered how one would implement this. Still seems like a looot of code for a very minor (?) change in error message that has to be maintained.

**Comment by orta** [comment]
Hrm, I think a simpler approach might be better here.

I think we should probably skip the saving to the home and just always use the API? I doubt we'd be causing enough traffic to be a problem to rubygems. This greatly simplifies the windows support, and the complexity tradeoff for the feature

**Comment by orta** [comment]
Considering I'm releasing multiple danger versions every week, there's _always_ a new version if you're using a `Gemfile`.

**Comment by JuanitoFatas** [comment]
> I think we should probably skip the saving to the home and just always use the API? I doubt we'd be causing enough traffic to be a problem to rubygems. This greatly simplifies the windows support, and the complexity tradeoff for the feature

Sounds good to me.

**Comment by hanneskaeufler** [comment]
Nice this looks like a great tradeoff!

**Comment by JuanitoFatas** [comment]
Apart [from 3 offesens](https://travis-ci.org/danger/danger/jobs/157203747#L957):

screenshot 2016-09-03 05 00 36

This PR is good to go.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * Fixes for `danger.import_dangerfile "org/repo"` - orta re:#487
 * Remove a variable assignment in comments_helper.rb - JuanitoFatas
+* Prints updated danger message when installed danger is outdated - JuanitoFatas
```
Comment (orta) [inline CHANGELOG.md]: can you move this up into the master section please ( https://github.com/dblock/danger-changelog/issues/9 )

**Comment by orta** [comment]
Alright looks good, I'd be alright with removing the assignment in `if` rubocop if you have strong opinions ( I've wrote that kind of code then sighed and removed it )

The exceptions can be easily switched that that should be good to go IMO

**Comment by orta** [comment]
Woooooooo

**Comment by orta** [comment]
Awesome, let's do it - thanks for the contribution!

**Pull request merged**